guide/src/telemetry.md

```diff
@@ -46,10 +46,11 @@ The following table describes the metrics currently tracked by the telemetry ser
 | `msg_num`                    | How many messages Hermes submitted to a specific chain. | `u64` Counter     |
 | `queries`                    | Number of queries emitted by the relayer, per chain and query type | `u64` Counter |
 | `query_cache_hits`           | Number of cache hits for queries emitted by the relayer, per chain and query type | `u64` Counter |
-| `send_packet_count`          | Number of SendPacket relayed | `u64` Counter |
-| `acknowledgement_count`      | Number of WriteAcknowledgement relayed               | `u64` Counter       |
-| `cleared_count`              | Number of SendPacket relayed through ClearPendingPackets | `u64` Counter   |
-| `oldest_sequence`            | The sequence number of the oldest pending SendPacket. If this value is 0, it means there are no pending SendPacket | `u64` ValueRecorder |
+| `send_packet_count`          | Number of SendPacket events processed | `u64` Counter |
+| `acknowledgement_count`      | Number of WriteAcknowledgement events processed      | `u64` Counter       |
+| `cleared_send_packet_count`    | Number of SendPacket events processed during the initial and periodic clearing | `u64` Counter   |
+| `cleared_acknowledgment_count` | Number of WriteAcknowledgement events processed during the initial and periodic clearing | `u64` Counter   |
```

Comment:
Could you try to get this counter in the example below? I tried and was not able to get it. Here is what I tried:
- send 20 packets to ibc-0
`hermes tx ft-transfer --sender-chain ibc-0 --receiver-chain ibc-1 --sender-channel channel-0 --sender-port transfer --amount 90 --timeout-height-offset 3000 --number-msgs 20 --key-name user2`
- check pending (shows 20 unreceived_packets)
`hermes  query packet pending --chain ibc-0 --channel channel-0 --port transfer`
- receive the 20 packets on ibc-1
`hermes tx packet-recv --receiver-chain ibc-1 --sender-chain ibc-0 --sender-port transfer --sender-channel channel-0`
- check pending (shows 20 unreceived_acks)
`hermes  query packet pending --chain ibc-0 --channel channel-0 --port transfer`
- send 10 more packets to ibc-0
`hermes tx ft-transfer --sender-chain ibc-0 --receiver-chain ibc-1 --sender-channel channel-0 --sender-port transfer --amount 90 --timeout-height-offset 3000 --number-msgs 10 --key-name user2`
- check pending (shows 10 unreceived_packets and 20 unreceived_acks)
`hermes  query packet pending --chain ibc-0 --channel channel-0 --port transfer`
- start hermes
`hermes start`
- wait a bit for hermes to finish clearing and then check metrics. I get these (cleared_acknowledgment_count):
```
# HELP acknowledgement_count Number of WriteAcknowledgement events processed
# TYPE acknowledgement_count counter
acknowledgement_count{chain="ibc-0",channel="channel-0",counterparty="ibc-1",port="transfer"} 10

# HELP cleared_send_packet_count Number of SendPacket events processed during the initial and periodic clearing
# TYPE cleared_send_packet_count counter
cleared_send_packet_count{chain="ibc-0",channel="channel-0",counterparty="ibc-1",port="transfer"} 10

# HELP msg_num How many messages Hermes submitted to the chain, per chain
# TYPE msg_num counter
msg_num{chain="ibc-0"} 32
msg_num{chain="ibc-1"} 11

# HELP send_packet_count Number of SendPacket events processed
# TYPE send_packet_count counter
send_packet_count{chain="ibc-0",channel="channel-0",counterparty="ibc-1",port="transfer"} 10 
```
For me also the latency metrics are missing for one chain (`ibc-1`)

Reply:
Indeed I am not getting the `cleared_acknowledgment_count` as expected. I will investigate this issue.

Reply:
There was an issue in where the `cleared_acknowledgment_count` was counted. I pushed a fix and running the example you gave now correctly has the following metrics:

```
acknowledgement_count{chain="ibc-0",channel="channel-0",counterparty="ibc-1",port="transfer"} 10
cleared_acknowledgment_count{chain="ibc-0",channel="channel-0",counterparty="ibc-1",port="transfer"} 20
send_packet_count{chain="ibc-0",channel="channel-0",counterparty="ibc-1",port="transfer"} 10
cleared_send_packet_count{chain="ibc-0",channel="channel-0",counterparty="ibc-1",port="transfer"} 10
```

There is indeed an issue recording latencies with the clearing:
* clearing unreceived_acks doesn't record any latency
* clearing unreceived_packets only record latencies for the sender chain
* sending packets when Hermes is running correctly record the latencies

I will check and see how this can be corrected

Reply:
excellent, thanks for looking into this!

Reply:
back to the counters, for the example I gave I get now (*):
```
acknowledgement_count{chain="ibc-0",channel="channel-0",counterparty="ibc-1",port="transfer"} 10
cleared_acknowledgment_count{chain="ibc-0",channel="channel-0",counterparty="ibc-1",port="transfer"} 20

send_packet_count{chain="ibc-0",channel="channel-0",counterparty="ibc-1",port="transfer"} 10
cleared_send_packet_count{chain="ibc-0",channel="channel-0",counterparty="ibc-1",port="transfer"} 10

msg_num{chain="ibc-0"} 32
msg_num{chain="ibc-1"} 11
```
should `acknowledgement_count` be 30 if it is the total? Meaning there were 20 pending acks plus the 10 that came in live from the processing of the 10 send_packet.

Reply:
For me this is confusing.  Also `send_packet_count` will not mirror `acknowledgment_count` if any packets have timed out. IMO we should be consistent for send and ack metrics. But if we keep things the way they are we need to explain very clearly what the metrics mean in the help strings. The name of the counters would also need to reflect the difference (`send_packet_count` is a total while the `acknowledgment_count` counts the live events only). Something like:

```
live_acknowledgement_count{chain="ibc-0",channel="channel-0",counterparty="ibc-1",port="transfer"} 10
cleared_acknowledgment_count{chain="ibc-0",channel="channel-0",counterparty="ibc-1",port="transfer"} 20

total_send_packet_count{chain="ibc-0",channel="channel-0",counterparty="ibc-1",port="transfer"} 10
cleared_send_packet_count{chain="ibc-0",channel="channel-0",counterparty="ibc-1",port="transfer"} 10
```

We could also have `live_send_packet_count` instead of `total`.

Reply:
Oh I hadn't taken into account the timeouts. The points raised regarding these metrics make me realise that these metrics seem more confusing than helpful.

The addition of these metrics was originally to spot slow/stalled traffic https://github.com/informalsystems/ibc-rs/issues/2175, but in my opinion the `oldest_sequence` and `oldest_timestamp` should be enough to track this.

Furthermore these metrics, `send_packet_count`, `acknowledgment_count`, `cleared_send_packet_count` and `cleared_acknowledgement_count` seem to aim at corner cases rather than general cases, they work well if there are no timeouts, and/or pending packets.

The more points were raised the more I had trouble convincing myself of the usefulness of these metrics. I am wondering if it is a good idea to keep them. What do you think ?

Reply:
I think we should keep something, they are helpful as there is no other hint on possible packets relayed when `tx_confirmation` is false. It's the best we can do to say we processed x, y type events. 
What is not completely clear is how to reduce the confusion. As I said, if we keep what we have, we can clarify by having good help and metric names. Since these are already in the code base I would say let's do it the best we can in this PR. Once we experiment more we can change later along the other lines I mentioned or something else or even remove them.